iree/compiler/Dialect/Stream/Transforms/DumpStatistics.cpp

```diff
@@ -0,0 +1,555 @@
+  prettyPrintSectionHeader("Constants / Variables", os);
+  os << "//\n";
+
+  // TODO(benvanik): print global information:
+  // - number of resource globals: constants/variables
+  // - util.byte_buffer.constant sizes (fed into stream.resource.try_map/map)
+  // - variable allocation sizes
+  os << "// TODO\n";
+
+  os << "//\n";
+}
+
+static void prettyPrintSyncInfo(const UsageInfo &usageInfo, bool verbose,
+                                llvm::raw_fd_ostream &os) {
+  prettyPrintSectionHeader("Synchronization", os);
+  os << "//\n";
+
+  // TODO(benvanik): print host <-> device information:
+  // - number of stream.timepoint.awaits
+  // - staging buffer allocation sizes
+  // - number of buffer mapping operations
+  // - estimated number of submissions (execution with await in the middle)
+  os << "// TODO\n";
+
+  os << "//\n";
+}
+
+static void prettyPrintStreamInfo(const UsageInfo &usageInfo,
+                                  IREE::Stream::CmdExecuteOp executeOp,
+                                  llvm::raw_fd_ostream &os) {
+  auto parentOp = executeOp->getParentWithTrait<mlir::OpTrait::FunctionLike>();
+
+  prettyPrintItemHeader(
+      llvm::formatv("stream.cmd.execute", parentOp->getName().getStringRef()),
+      os);
+  os << "// ";
+  prettyPrintOpBreadcrumb(executeOp, os);
+  os << "\n";
+  os << "//\n";
+
+  // TODO(benvanik): print stream information (for each stream.cmd.execute):
+  // - number of unique resources captured
+  // - number of commands of each type
+  // - % concurrently executable
+  os << "// TODO\n";
+}
+
+static void prettyPrintAllStreamInfo(const UsageInfo &usageInfo, bool verbose,
+                                     llvm::raw_fd_ostream &os) {
+  prettyPrintSectionHeader("Streams", os);
+  os << "//\n";
+
+  // TODO(benvanik): aggregate stats:
+  // - number of streams
+  // - (eventually) number of streams per affinity
+  // - average commands per stream
+  // - streams with host dependencies/device dependencies (awaits/etc)
+  os << "// TODO\n";
+
+  os << "//\n";
+  for (auto executeOp : usageInfo.executeOps) {
+    prettyPrintStreamInfo(usageInfo, executeOp, os);
+    os << "//\n";
+  }
+}
+
+static void prettyPrintExecutableExportInfo(
+    const UsageInfo &usageInfo, IREE::Stream::ExecutableOp executableOp,
+    IREE::Stream::ExecutableExportOp exportOp, llvm::raw_fd_ostream &os) {
+  auto funcOp = exportOp.getFunctionRef();
+  prettyPrintItemHeader(
+      llvm::formatv("stream.executable.export @{0}::@{1}",
+                    executableOp.getName(), exportOp.getName()),
+      os);
+  os << "// ";
+  prettyPrintOpBreadcrumb(funcOp, os);
+  os << "//\n";
+  os << "//\n";
+
+  // TODO(benvanik): interface and usage stats:
+  // - operand info
+  // - binding info
+  //   - misaligned/unaligned/etc - big warning
+  // - incoming dispatches
+  //   - workload params
+
+  // TODO(benvanik): ask codegen team if they want anything like a list of
+  // linalg named ops, etc.
```

Comment:
This seems similar to https://github.com/google/iree/issues/7211

Reply:
somewhat - better naming here will definitely help things be more readable but having the linalg named ops listed (I think that can be done with an interface) is different as we'd be able to have some better output (can't serialize a set of root ops with all their sizes/index maps/etc into a symbol name)